src/drivers/imu/adis16477/ADIS16477.cpp

```diff
@@ -529,19 +454,38 @@ ADIS16477::write_reg16(uint8_t reg, uint16_t value)
 void
 ADIS16477::start()
 {
+#ifdef GPIO_SPI1_DRDY1_ADIS16477
+	// setup data ready on rising edge
+	stm32_gpiosetevent(GPIO_SPI1_DRDY1_ADIS16477, true, false, true, &ADIS16477::data_ready_interrupt, this);
+#else
 	/* make sure we are stopped first */
-	uint32_t last_call_interval = _call_interval;
 	stop();
-	_call_interval = last_call_interval;
 
 	/* start polling at the specified rate */
 	hrt_call_every(&_call, 1000, _call_interval, (hrt_callout)&ADIS16477::measure_trampoline, this);
+#endif
 }
 
 void
 ADIS16477::stop()
 {
+#ifdef GPIO_SPI1_DRDY1_ADIS16477
+	// stop data ready
+	stm32_gpiosetevent(GPIO_SPI1_DRDY1_ADIS16477, false, false, false, nullptr, nullptr);
```

Comment:
micro hal https://github.com/PX4/Firmware/blob/master/src/platforms/px4_micro_hal.h#L216

Reply:
px4_arch_gpiosetevent is one argument short, need to review.

Reply:
Well there is a PR opportunity for you! I suspect it is the addition of `this` in nuttx's IRQ dispatch